deployment/environment/devenv/environment.go

```diff
@@ -39,7 +39,7 @@ func NewEnvironment(ctx context.Context, lggr logger.Logger, config EnvironmentC
 	}
 	var nodeIDs []string
 	if jd.don != nil {
-		err = jd.don.CreateSupportedChains(ctx, config.Chains, *jd)
+		err = jd.don.CreateSupportedChains(ctx(), config.Chains, *jd)
```

Comment:
This is fine as-is, but declaring one `ctx := ctx()` at the top of the func can be less error prone over time.

Reply:
Ah, but this is a more complicated case where the context is being used in the constructor rather than a method. This is often problematic because it leaves the user unable to control the two separately. I think it would be appropriate to pass a context normally to these calls that use it synchronously. But OFC that would be messy alongside the `func() context.Context`, so probably only along with https://github.com/smartcontractkit/chainlink/pull/15410/files#r1859485133

Reply:
I agree in principle - my plan was to fix those constructors in a subsequent PR, where possible (or use the mentioned pattern). This was an enabling change for some other situations where contexts are being used, but naively in the middle of downstream code in this and chainlink-deployments, and I wanted the plumbing all the way through, to at least make sure we HAD cancellation, and had a way to avoid folks using weird lambdas (not like what you're suggesting, but as complete workarounds of the outer API, to plumb through things like the context, and certain secrets, etc.).

Now that I understand context better, I can do a cleanup of context more generally of code in /development and chainlink-deployments.